File: app/src/main/java/org/coscup/ccip/adapter/ScheduleTabAdapter.java
package org.coscup.ccip.adapter;

import android.support.v4.app.Fragment;
import android.support.v4.app.FragmentManager;
import android.support.v4.app.FragmentStatePagerAdapter;

import java.util.ArrayList;
import java.util.List;

import org.coscup.ccip.fragment.ScheduleFragment;

public class ScheduleTabAdapter extends FragmentStatePagerAdapter {
    private final List<Fragment> mFragmentList = new ArrayList<>();
    private final List<String> mFragmentTitleList = new ArrayList<>();

    public ScheduleTabAdapter(FragmentManager fm) {
        super(fm);
    }

    @Override
    public Fragment getItem(int position) {
        return mFragmentList.get(position);
    }

    @Override
    public int getCount() {
        return mFragmentList.size();
    }

    @Override
    public CharSequence getPageTitle(int position) {
        return mFragmentTitleList.get(position);
    }

    public void addFragment(Fragment fragment, String title) {
        mFragmentList.add(fragment);
        mFragmentTitleList.add(title);
    }

    public void toggleStarFilter(boolean isStar) {
        for (Fragment fragment : mFragmentList) {
            ((ScheduleFragment) fragment).toggleStarFilter(isStar);
        }
    }
}

File: app/src/main/java/app/opass/ccip/ui/schedule/ScheduleTabAdapter.kt
package app.opass.ccip.ui.schedule

import androidx.fragment.app.FragmentManager
import androidx.fragment.app.FragmentStatePagerAdapter

class ScheduleTabAdapter(
    fm: FragmentManager,
    private val dates: List<String>
) : FragmentStatePagerAdapter(fm, BEHAVIOR_RESUME_ONLY_CURRENT_FRAGMENT) {
    override fun getItem(position: Int) = ScheduleFragment.newInstance(dates[position])

    override fun getCount() = dates.size

    override fun getPageTitle(position: Int) = dates[position]
}
